### PR TITLE
feat: add TS coverage report per package

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

This PR creates a TS report per package after running yarn ts-coverage at the root of the workspace or at every package.

The following changes are:
-Adds current changes from [https://github.com/aws-amplify/amplify-js/pull/10293](url)
-Adds ts-coverage script to every package
-Adds lerna ts-coverage script at the root of the workspace 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/tsconfig.build.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"extends": "../tsconfig.settings.json",
+	"compilerOptions": {
+		"composite": true
```
Comment (jimblanc) [inline packages/amazon-cognito-identity-js/tsconfig.build.json:4]: Should this be moved up to `tsconfig.settings.json`? Seems like it appears in every package

File: scripts/build.js [hunk 2]
```diff
@@ -153,32 +157,24 @@ async function buildES5(typeScriptCompiler, watchMode) {
...
-		incremental: true,
+		jsx: jsx,
 		tsBuildInfoFile: es5TsBuildInfoFilePath,
 		typeRoots,
-		// temporary fix
-		types: ['node'],
 		outDir: pkgTscES5OutDir,
 	};
 
+	if ('composite' in compilerOptions) delete compilerOptions['composite'];
```
Comment (jimblanc) [inline scripts/build.js:176]: Could you add a small comment explaining why we're doing this?

**Review by jimblanc — COMMENTED** [review]
LGTM besides some nitpicks, thanks for setting this up!

File: packages/amazon-cognito-identity-js/tsconfig.build.json (hunk 1 above)
Comment (israx) [reply packages/amazon-cognito-identity-js/tsconfig.build.json:4]: Actually, I'm removing this "composite" flag, there's not use to it

File: scripts/build.js (hunk 2 above)
Comment (israx) [reply scripts/build.js:176]: As I'm removing the "composite" flag, that code is not needed any more

**Review by jimblanc — APPROVED** [review]

**Pull request merged**